Sources/CircularProgress/protocol/MouseTrackable.swift

```diff
@@ -0,0 +1,25 @@
+#if os(macOS)
+import Cocoa
+/**
+Makes it possible to utilize mouseOver/mouseOut functionality on a view (Supports custom bezier path)
+*/
+protocol MouseTrackable:class{
+	var path:CGPath? {get}
+}
+extension MouseTrackable where Self:NSView{
+	/**
+	* Creates a new tracking area
+	*/
+	internal func createTrackingArea(_ options:NSTrackingArea.Options = [.activeInActiveApp, .mouseEnteredAndExited, .mouseMoved]){
+		trackingAreas.forEach{ trackingArea in
+			self.removeTrackingArea(trackingArea)
+		}
+		let trackingArea:NSTrackingArea = .init(rect: self.bounds, options: options, owner: self, userInfo: nil)
+		self.addTrackingArea(trackingArea)
+	}
+	/**
+	Stores/generates the path to hitTest against
+	*/
+	var path:CGPath? {return nil}
```

Comment:
The naming here is too generic and could easily conflict. Should be `hitTestPath` or something better.

Reply:
Why limit it to a single path. Seems like if we want to make it generally useful, it should be an array of paths?

Reply:
Im just solving for the CircularProgress project here. I also have this "tech" in another OSS project https://github.com/eonist/Hybrid/blob/master/src/button/button/Button%2BTrackable.swift , were it can gain more things like this. Maybe even become its own Component or maybe a general tutorial on how to do over/Out on custom paths. Which should be so simple 🤷, but isn't 🙁

Reply:
Ill fix the name. Let me know if you want array support on paths 👍

Reply:
Fixed the name. ✅ Not doing the multiple path support. Unless you are sure you want it. Let me know. It's  pretty simple.